common/src/main/java/io/netty/util/concurrent/ScheduledFutureTask.java

```diff
@@ -78,12 +83,17 @@ protected EventExecutor executor() {
         return super.executor();
     }
 
+    @Override
     public long deadlineNanos() {
         return deadlineNanos;
     }
 
     public long delayNanos() {
-        return Math.max(0, deadlineNanos() - nanoTime());
+        return deadlineToDelayNanos(deadlineNanos());
+    }
+
+    static long deadlineToDelayNanos(long deadlineNanos) {
+        return Math.max(0, deadlineNanos - nanoTime());
```

Comment:
I agree this could likely be changed, but is pre-existing behavior and doesn't impact this change. Note that we are sorting, and determining "next to expire", based upon deadline (not delay).

Lets discuss in a followup issue: https://github.com/netty/netty/issues/7840